python/py-awscli/Portfile

```diff
@@ -52,5 +53,24 @@ if {${name} ne ${subport}} {
              ${worksrcpath}/${subport}
     }
 
+    variant bash_completion {
+        post-destroot {
+            set bash_compl_path ${prefix}/etc/bash_completion.d
+            xinstall -d ${destroot}${bash_compl_path}
+            xinstall -m 0644 ${worksrcpath}/bin/aws_bash_completer \
+                ${destroot}${bash_compl_path}/aws_bash_completer
+        }
+    }
```

Comment:
Generally, we just include completion scripts for our ports without a variant because they're tiny. I'd suggest just unconditionally installing these instead of having a variant.

Reply:
I believe if a variant is not used, then Macports will complain when installing, say, `py27-awscli`, after having already installed `py36-awscli` since the completion files will be owned by the already installed port.

Reply:
You could just give the autocomplete files names with the subport version included somewhere in order to avoid the clash, say "py36-aws_bash_completer".

BTW, it occurs to me that given that the aws cli is not a library, it might not belong in python/ as such. But that's another story and you can ignore it for now.

Reply:
> You could just give the autocomplete files names with the subport version included somewhere in order to avoid the clash, say "py36-aws_bash_completer".

Whilst that will probably work for bash, the zsh file has to be named `_aws` in order to work.

> BTW, it occurs to me that given that the aws cli is not a library, it might not belong in python/ as such.

I get that, though I do like that it makes it easy to run it with different interpreters.

Reply:
Then just create a `zsh` variant and install the bash completion file by default? More people use bash anyway.

Reply:
Makes sense. I'll give that a go.

By the way, is there an easy way to prevent binaries being installed? I'm asking because this port installs a number of files to `${prefix}/bin` which needn't be there: 

* `aws.cmd` (Windows only)
* `aws_bash_completer` (useless in that directory)
* `aws_zsh_completer.sh` (useless in that directory)

Reply:
Patch out the install?

Reply:
one easy way is to delete what you don't want installed in a `pre-destroot` block.

Reply:
Thanks, both. I'll take a look at that later today. It's not a big deal. 

I've pushed the change now, so only zsh_completion is a separate variant.